tests/testthat/test-vimpact.R

```diff
@@ -214,3 +214,90 @@ test_that("can get FVPS", {
     dplyr::arrange(activity_type, year, age)
   expect_equal(new_fvps, old_fvps, ignore_attr = TRUE)
 })
+
+test_that("can use wrapper function to run impact for a recipe", {
+  mock_data_1 <- data.frame(
+    country = c("AFG", "AFG"),
+    vaccine = c("HepB", "HepB"),
+    activity_type = c("routine", "routine"),
+    year = c(2000, 2001),
+    burden_outcome = c("dalys", "dalys"),
+    impact = c(2342, 3543)
+  )
+  mock_data_2 <- data.frame(
+    country = c("AGO", "AGO"),
+    vaccine = c("YF", "YF"),
+    activity_type = c("campaign", "campaign"),
+    year = c(2000, 2001),
+    burden_outcome = c("dalys", "dalys"),
+    impact = c(653, 765)
+  )
+  mock_calculate_impact <- mockery::mock(mock_data_1, mock_data_2,
+                                         cycle = TRUE)
+
+  recipe <- data.frame(
+    touchstone = c("201710gavi-5", "201710gavi-6"),
+    modelling_group = c("CDA-Razavi", "XYZ-someone"),
+    disease = c("HepB", "YF"),
+    focal = c("default:HepB_BD-routine;HepB-routine",
+              "default:YF-routine;YF-campaign"),
+    baseline = c("novac", "novac:YF-routine"),
+    burden_outcome = c("hepb_deaths_acute;hepb_cases_acute_severe", "*"))
+  t <- tempfile(fileext = ".csv")
+  write.csv(recipe, t, row.names = FALSE)
+
+  with_mock(calculate_impact = mock_calculate_impact, {
```

Comment:
The "current" way of doing this is to use `mockery::stub`, see for an example https://github.com/mrc-ide/dust/blob/568f4ecfac6f674955b27a22daf7b00de6b0fc84/tests/testthat/test-gpu.R#L99-L112

Reply:
When running interactively `mockery::stub` creates a function in the global environment with same name as the package function meaning that if I do `devtools::load_all` and try to re-run the test interactively it will still run the old version of the function. So using `with_mock` supports doing
1. Run test interactively - notice a failure
2. Update impl to fix test failure
3. `devtools::load_all()`
4. Run test interactively again

Using `stub` you have to restart R session between 2 & 3 which is a bit annoying

Reply:
I've got a workflow for this that I can show you, the trick is just not to evaluate the test in the global context